### PR TITLE
Edit pass for azeventhubs readme

### PR DESCRIPTION
**Pull request by bobtabor-msft**
AB#55131

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by bobtabor-msft** [comment]
#add-reviewer: @sandeep-sen

File: sdk/messaging/azeventhubs/README.md [hunk 1]
```diff
@@ -23,41 +28,35 @@ Install the Azure Event Hubs client module for Go with `go get`:
...
 
-Events can be consumed from an event hub using an [event consumer](https://docs.microsoft.com/azure/event-hubs/event-hubs-features#event-consumers). In this package there are two types for consuming events: 
-- The basic event consumer is the, in the [ConsumerClient](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#ConsumerClient). This consumer is useful if you already known which partitions you want to receive from.
-- A distributed event consumer, which uses Azure Blobs for checkpointing and coordination. This is implemented in the [Processor](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#Processor). This is useful when you want to have the partition assignment be dynamically chosen, and balanced with other Processor instances.
+Events can be consumed from an event hub using an [event consumer][eventhubs_eventconsumer]. In this package there are two types for consuming events: 
+- The basic event consumer is the, in the [ConsumerClient][azeventhubs_consumerclient]. This consumer is useful if you already known which partitions you want to receive from.
```
Comment (bobtabor-msft) [inline sdk/messaging/azeventhubs/README.md:52]: "The basic event consumer is the," ... the what?

File: sdk/messaging/azeventhubs/README.md [hunk 2]
```diff
@@ -23,41 +28,35 @@ Install the Azure Event Hubs client module for Go with `go get`:
...
-### Prerequisites
-
-- Go, version 1.18 or higher
-- An [Azure subscription](https://azure.microsoft.com/free/)
-- An [Event Hub namespace](https://docs.microsoft.com/azure/event-hubs/).
-- An Event Hub. You can create an event hub in your Event Hubs Namespace using the [Azure Portal](https://docs.microsoft.com/azure/event-hubs/event-hubs-create), or the [Azure CLI](https://docs.microsoft.com/azure/event-hubs/event-hubs-quickstart-cli).
-
 ### Authenticate the client
 
 Event Hub clients are created using an Event Hub a credential from the [Azure Identity package][azure_identity_pkg], like [DefaultAzureCredential][default_azure_credential].
```
Comment (bobtabor-msft) [inline sdk/messaging/azeventhubs/README.md:33]: I would like to re-write this entire H3's content (including the H4s) to fix some grammar (passive voice in the first sentence), to describe how to authenticate more detail, and to provide more context to the code examples. However, I don't have the subject matter expertise to reword this section. Please think about editing this for clarity. Thank you!

File: sdk/messaging/azeventhubs/README.md [hunk 3]
```diff
@@ -98,36 +97,63 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
+[eventhubs_docs]: https://azure.microsoft.com/services/event-hubs/
+[eventhubs_eventpublisher]: https://docs.microsoft.com/azure/event-hubs/event-hubs-features#event-publishers
+[eventhubs_features_terminology]: https://docs.microsoft.com/azure/event-hubs/event-hubs-features
+[eventhubs_namespace]: https://docs.microsoft.com/azure/event-hubs/event-hubs-features#namespace
+[eventhubs_partition]: https://docs.microsoft.com/azure/event-hubs/event-hubs-features#partitions
+[eventhubs_rest_docs]: https://learn.microsoft.com/en-us/rest/api/eventhub/
+[go_install]: https://go.dev/doc/install
+[security_techcenter]: https://www.microsoft.com/msrc/faqs-report-an-issue
 
 ![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-go%2Fsdk%2Fmessaging%2Fazeventhubs%2FREADME.png)
```
Comment (bobtabor-msft) [inline sdk/messaging/azeventhubs/README.md:159]: Curious what this is. It does not exist in the other document that Dan reviewed: https://github.com/Azure/azure-sdk-for-go/pull/20382/commits/3a5bcc1c2cf8b19be181343952f13bc00d4497e9

File: sdk/messaging/azeventhubs/README.md [hunk 4]
```diff
@@ -98,36 +97,63 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
 a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
```
Comment (bobtabor-msft) [inline sdk/messaging/azeventhubs/README.md:97]: The Contributing section is different from what is in this document: https://github.com/Azure/azure-sdk-for-go/pull/20382/commits/3a5bcc1c2cf8b19be181343952f13bc00d4497e9

Consider standardizing?

File: sdk/messaging/azeventhubs/README.md [hunk 5]
```diff
@@ -98,36 +97,63 @@ When you submit a pull request, a CLA-bot will automatically determine whether y
...
-
-[azure_identity_pkg]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity
+[azure_sdk_for_go_wiki]: https://github.com/azure/azure-sdk-for-go/wiki
+[azure_sub]: https://azure.microsoft.com/free/
+[azeventhubs_connectionstring_consumerclient]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#example-NewConsumerClientFromConnectionString
+[azeventhubs_connectionstring_producerclient]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#example-NewProducerClientFromConnectionString
+[azeventhubs_consumerclient]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs#ConsumerClient
+[azeventhubs_mit_license]: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs/LICENSE.txt
+[azeventhubs_repo]: https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/messaging/azeventhubs
+[azeventhubs_repo_main]: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs
```
Comment (bobtabor-msft) [inline sdk/messaging/azeventhubs/README.md:135]: Should these two be the same? They're in different parts of the document ... but shouldn't they be pointing to the same URL?  If not, please ignore. Thank you!

**Review by bobtabor-msft — COMMENTED** [review]
@sandeep-sen In my review, I found a few potential issues that I can't address yet in my tenure as your docs writer due to lack of subject matter expertise. Please feel free to ignore / resolve.

File: sdk/messaging/azeventhubs/README.md (hunk 1 above)
Comment (bobtabor-msft) [reply sdk/messaging/azeventhubs/README.md:52]: Also, not to self ... missed "if you already known which".

Comment (bobtabor-msft) [reply sdk/messaging/azeventhubs/README.md:52]: @richardpark-msft Can you address this?

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/README.md:52]:
```suggestion
- The basic event consumer is the [ConsumerClient][azeventhubs_consumerclient]. This consumer is useful if you already known which partitions you want to receive from.
```

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/README.md:52]: @bobtabor-msft, does this edit sound okay?

**Review by sandeep-sen — APPROVED** [review]

**Pull request closed without merging**